File: backend/GymRecommender/models/notification.py
from django.db import models
from .account import Account
from ..enums.notification_type import NotificationType

class Notification(models.Model):
    id = models.UUIDField(primary_key=True)
    type = models.TextField(choices=NotificationType)
    message = models.TextField()
    created_at = models.DateTimeField()
    read_at = models.DateTimeField(blank=True, null=True)
    user = models.ForeignKey(Account, models.DO_NOTHING)

    class Meta:
        managed = False
        db_table = 'notification'

File: backend/Models/Notification.cs
using System;
using System.Collections.Generic;
using backend.Enums;

namespace backend.Models;

public partial class Notification
{
    public Guid Id { get; set; }

    public string Message { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime? ReadAt { get; set; }
    
    public NotificationType Type { get; set; }

    public Guid UserId { get; set; }

    public virtual Account User { get; set; } = null!;
}
